contracts/carbon/Strategies.sol

```diff
@@ -514,17 +522,23 @@ abstract contract Strategies is Initializable {
     /**
      * @dev calculates the required amount plus fee
      */
-    function _addFee(uint128 amount) private view returns (uint128) {
+    function _addFee(uint128 amount, uint128 pairId) private view returns (uint128) {
+        // override protocol-wide trading fee with custom one if it's set for the pair
+        uint32 tradingFeePPMOverrides = _tradingFeePPMOverrides[pairId];
+        uint32 tradingFeePPM = tradingFeePPMOverrides == 0 ? _tradingFeePPM : tradingFeePPMOverrides;
         // divide the input amount by `1 - fee`
-        return MathEx.mulDivC(amount, PPM_RESOLUTION, PPM_RESOLUTION - _tradingFeePPM).toUint128();
+        return MathEx.mulDivC(amount, PPM_RESOLUTION, PPM_RESOLUTION - tradingFeePPM).toUint128();
     }
 
     /**
      * @dev calculates the expected amount minus fee
      */
-    function _subtractFee(uint128 amount) private view returns (uint128) {
+    function _subtractFee(uint128 amount, uint128 pairId) private view returns (uint128) {
+        // override protocol-wide trading fee with custom one if it's set for the pair
+        uint32 tradingFeePPMOverrides = _tradingFeePPMOverrides[pairId];
+        uint32 tradingFeePPM = tradingFeePPMOverrides == 0 ? _tradingFeePPM : tradingFeePPMOverrides;
```

Comment:
Suggesting to reuse this code in a single (private) helper function.
Especially in light of the fact that bytecode size is close to the limit.

Reply:
Extracted repeating logic in a helper function.